beluga/routes.py

```diff
@@ -66,7 +70,78 @@
     @apiSuccess {String} next URL of the next page of results.
     @apiSuccess {String} previous URL of the previous page of results.
     """
-    raise abort(501, 'not implemented')
+
+    start_time = request.args.get('start_time')
+    end_time = request.args.get('end_time')
+    lat = request.args.get('lat')
+    lon = request.args.get('lon')
+    radius = request.args.get('radius')
+
+    # Validate the parameters we got, if any
+    if start_time or end_time:
+        if not start_time or not end_time:
+            raise abort(400, "temporal range must be fully specified")
+
+    if lat or lon or radius:
+        if not lat or not lon or not radius:
+            raise abort(400, "search area must be fully specified")
+    
+    # Acquire the events the user requested
+    with session_scope() as db_session:
+        events = db_session.query(Event)
+        if start_time:
+            events = events.filter(Event.start_time >= start_time)
+            events = events.filter(Event.end_time <= end_time)
+        if lat:
+            # Convert from lat/lon/radius to upper/lower bounds for lat/lon
+
+            # We need to find the length of one degree longitude at the
+            # center point we were given. To do so, we take the cosine
+            # of the latitude we are given and multiply that by the length
+            # of one degree longitude at the equator.
+            lat_rads = (math.pi * float(lat)) / 180.0 # given lat in radians
+            lon_factor = 1.0 / ((math.pi / 180.0) * 6371.0 * math.cos(lat_rads))
```

Comment:
Can we pull `6371.0` out into a variable?

Reply:
yep, good call